### ISSUE TITLE
`userId` distinguished attributes available with passTLSClientCert infos

### ISSUE THREAD

**Issue by GKTheOne** [traefik/traefik#4391]
### Do you want to request a *feature* or report a *bug*?

Feature

### What did you expect to see?

As per #4199, but I want 'userId'. (If I can't have the whole subject)

**Comment by tpdownes** [comment]
I'm not sure what `userId` is but I agree that the implementation here is weird, randomly picking out bits of the subject rather than just giving it to you. Here's a link to the current standard:

https://tools.ietf.org/html/rfc5280#section-4.1.2.6

which says that certificate subjects must comply with same rules as issuer subjects:

https://tools.ietf.org/html/rfc5280#section-4.1.2.4

**Comment by GKTheOne** [comment]
@tpdownes I was not making any comments on the implementation, however I will say that it is slightly troublesome because we've had to add X509 (and base64) decoding to an application just to get unsupported (by traefik) attribute type values from the client certificate subject. Previously we could have used our F5 proxy to put the subject string into a header.

I will happily contribute to another issue (is there one?) where we can discuss ideal implementations, but I don't want to get bogged down in discussions about those ideal implementations here.

Since there is precedent work as part of #4298 and because I want it "now", I am hoping that the `userId` attribute type can be added to the list of known attribute types processed by traefik.

- short name: `userId`
- long name: `uid`
- object id: `0.9.2342.19200300.100.1.1`

(I believe the latest definition is in [RFC4519](https://tools.ietf.org/html/rfc4519#section-2.39) as mentioned by [RFC4524](https://tools.ietf.org/html/rfc4524#section-1.1))

**Comment by tpdownes** [comment]
My point is that `userID` isn't, so far as I can tell, a required attribute of X509 subjects even if it's part of the X.500 scheme. I do see UID in Apache support for header info, though so maybe it's ubiquitous in practice.

I'm not trying to hijack the issue -- and appreciate Traefik handling my issue promptly as I hope they do yours. But they could have just implemented the list of [Apache](https://httpd.apache.org/docs/current/mod/mod_ssl.html#envvars) headers I linked to, and your problem would already have been solved.
```
x509 specifies a component of an X.509 DN; one of C,ST,L,O,OU,CN,T,I,G,S,D,UID,Email
```
(some of which often appear multiple times in a subject)

Even with the added support, I too have to send the full PEM to my backend.
